packages/kit/src/runtime/server/page/load_node.js

```diff
@@ -52,13 +52,15 @@ export async function load_node({
 	/** @type {import('types').LoadOutput} */
 	let loaded;
 
+	const should_prerender = node.module.prerender ?? state?.prerender?.default ?? false;
```

Comment:
"should" is very vague. it doesn't really clarify under what conditions prerendering happens. I think `prerender_enabled` would make it clearer that we're checking whether it's enabled for the page

I also realized this could be written as `!!(node.module.prerender ?? state?.prerender?.default)` which might be easier to read by removing the double nullish coalescing?

Reply:
I'm just copying the naming from [here](https://github.com/sveltejs/kit/blob/39eaf0905f28c55ba7ee3fe9ba67ec487290a6b2/packages/kit/src/runtime/server/page/respond.js#L75) for consistency:

```ts
const should_prerender = leaf.prerender ?? state.prerender.default;
```

If you want to go with `prerender_enabled`, I'm happy to make that change! Just want to confirm before we name it inconsistently.

As for the double-bang, it's funny -- I work a _lot_ in C# and Go. In the process of opening the 3-4 PRs I currently have going, I actually had to Google what the heck the double-bang was doing, as the coalescing chain is what I'm used to! 😆 If it's clearer to JS developers to use the double-bang, I'll change it along with the variable name.

Reply:
@benmccann - I just had one of those "eyes pop open in the middle of the night" moments about this one. Doing something completely unrelated, hadn't thought about this in days, and there it was -- like some ugly toad jumping out of the grass on an evening walk. The reason it feels weird to name the variable `prerender_enabled` is because that's _not_ what the variable means. There's a setting in config, `prerender.enabled`, which controls whether Kit even _tries_ to prerender your app. I believe that flag is available in this context as `state.prerender.enabled`. This check is for whether or not this node _should_ prerender, which is determined by the combination of the node's exported `prerender` attribute and the config's prerendering default.

Reply:
I know it's not referring to whether it's enabled at the application level, but it does refer to whether it's enabled at the page level. Anyway, I'll let Rich weigh in with his thoughts

Reply:
i think `should_prerender` is fine — i agree that `prerender_enabled` sounds too much like it's just `config.kit.prerender.enabled`